### PR TITLE
Simplify permission check

### PR DESCRIPTION
**Pull request by mathiascode**

### CODE REVIEW

**Review by madmaxoft — APPROVED** [review]

**Comment by madmaxoft** [comment]
I'd like someone else to take a look at this, too.

File: app/src/main/java/org/cuberite/android/MainActivity.java [hunk 1]
```diff
@@ -107,35 +107,33 @@ public static void showSnackBar(Activity activity, String message) {
...
         permissionPopup.show();
     }
 
     private void checkPermissions() {
-        if (preferences.getString("cuberiteLocation", null) == null) {
-            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
-                // Always use public dir in Lollipop and earlier, since permissions are granted when the app is installed
-                preferences.edit().putString("cuberiteLocation", PUBLIC_DIR + "/cuberite-server").apply();
-            } else {
+        if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
```
Comment (ItsShadowCone) [inline app/src/main/java/org/cuberite/android/MainActivity.java:128]: I do not like the reverse order of the `if` statements here, on Android <= 6 they lead to an unnecessary file write on every single `checkPermissions()` call, which is quite sprawled all over the place.

Two solutions would be either reverting to the not-null check before everything else or alternatively check in the else branch below ([L 137](https://github.com/cuberite/android/pull/39/files#diff-53dcaef8d195b3ca85811c12b44f377eR137)) whether the preferences show the correct location before writing to the preferences file.

**Review by ItsShadowCone — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/cuberite/android/MainActivity.java (hunk 1 above)
Comment (mathiascode) [reply app/src/main/java/org/cuberite/android/MainActivity.java:128]: Good point! I don't remember how it used to be, but checkPermissions is only called when the app resumes.

File: app/src/main/java/org/cuberite/android/MainActivity.java [hunk 2]
```diff
@@ -107,33 +107,33 @@ public static void showSnackBar(Activity activity, String message) {
...
-                    Log.d(LOG, "Requesting permissions for external storage");
-                    ActivityCompat.requestPermissions(MainActivity.this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, REQUEST_WRITE_EXTERNAL_STORAGE_PERMISSION);
-                }
-            })
-            .create();
-
-        permissionPopup.show();
+    private void showPermissionPopup() {
+        if (permissionPopup == null) {
+            permissionPopup = new AlertDialog.Builder(this)
```
Comment (ItsShadowCone) [inline app/src/main/java/org/cuberite/android/MainActivity.java]: I believe `permissionPopup` is never set back to `null` or am i wrong?

File: app/src/main/java/org/cuberite/android/MainActivity.java [hunk 3]
```diff
@@ -107,33 +107,33 @@ public static void showSnackBar(Activity activity, String message) {
...
+                .setTitle(getString(R.string.status_permissions_needed))
+                .setMessage(R.string.message_externalstorage_permission)
+                .setCancelable(false)
+                .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int id) {
+                        Log.d(LOG, "Requesting permissions for external storage");
+                        ActivityCompat.requestPermissions(MainActivity.this, new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE}, REQUEST_WRITE_EXTERNAL_STORAGE_PERMISSION);
```
Comment (ItsShadowCone) [inline app/src/main/java/org/cuberite/android/MainActivity.java]: Consider adding a `permissionPopup = null;` in here (i'd believe)

File: app/src/main/java/org/cuberite/android/MainActivity.java [hunk 4]
```diff
@@ -152,6 +150,14 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
             }
         }
     }
+    
+    @Override
+    public void onPause() {
+        super.onPause();
+        if (permissionPopup != null) {
+            permissionPopup.dismiss();
```
Comment (Cl1608Ho) [inline app/src/main/java/org/cuberite/android/MainActivity.java]: Also set `permissionPopup = null` here, otherwise we might dismiss a popup twice...

**Pull request merged**